flag_engine/segments/evaluator.py

```diff
@@ -28,51 +34,166 @@ def get_identity_segments(
     environment: EnvironmentModel,
 ) -> typing.List[SegmentModel]:
     """
-    Get a list of segments for a given identity in a given environment.
+    DEPRECATED: Get a list of segments for a given identity in a given environment.
 
     :param identity: the identity model object to get the segments for
     :param environment: the environment model object the identity belongs to
     :return: list of segments that the identity belongs to in the environment
     """
+    warnings.warn(
+        "`get_identity_segments` is deprecated, use `get_evaluation_result` instead.",
+        DeprecationWarning,
+    )
     context = map_environment_identity_to_context(
         environment=environment,
         identity=identity,
         override_traits=None,
     )
-    return get_context_segments(context, environment.project.segments)
+    return [
+        SegmentModel(id=segment_context["key"] or 0, name=segment_context["name"])
```

Comment:
I'm confused about the typing, given `SegmentContext.key` [expects a str](https://github.com/Flagsmith/flagsmith-engine/blob/fb31c9c463a5068c2ffa5149a8a219eb77cb7d0b/flag_engine/context/types.py#L53).